np.bincount broken on master

#1249 breaks np.bincount on GCC-5. Here is a short reproducing example:

```
import numpy as np

#pythran export compute(int8[:])
def compute(vector):
   return np.bincount(vector)
```

And the Travis logs for a slightly longer version:
int8: https://travis-ci.org/Dapid/test_travis/jobs/511931148 
int32: https://travis-ci.org/Dapid/test_travis/jobs/511932347

And only on int8 for the shown example: https://travis-ci.org/Dapid/test_travis/jobs/511933665
The error is:

```
/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/pythran/pythonic/numpy/bincount.hpp:22:22: error: no matching function for call to ‘max(long int&, int)’
     length = std::max(length, 1 + max(expr));
```

On my machine I have  GCC 8.3.1, and that works.

--- a/pythran/pythonic/numpy/bincount.hpp
+++ b/pythran/pythonic/numpy/bincount.hpp
@@ -19,7 +19,7 @@ namespace numpy
     long length = 0;
     if (minlength)
       length = (long)minlength;
-    length = std::max(length, 1 + max(expr));
+    length = std::max<long>(length, 1 + max(expr));
     types::ndarray<long, types::pshape<long>> out(types::pshape<long>(length),
                                                   0L);
     for (auto iter = expr.fbegin(), end = expr.fend(); iter != end; ++iter)
@@ -39,7 +39,7 @@ namespace numpy
     long length = 0;
     if (minlength)
       length = (long)minlength;
-    length = std::max(length, 1 + max(expr));
+    length = std::max<long>(length, 1 + max(expr));
     typename std::enable_if<
         std::tuple_size<pS>::value == 1,
         types::ndarray<decltype(std::declval<long>() *

